Compilation issue on reactor 3.0.4 upgrade

master has this compilation issue:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project spring-cloud-stream-reactive: Compilation failure
[ERROR] /home/travis/build/spring-cloud/spring-cloud-stream/spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/ObservableToMessageChannelResultAdapter.java:[50,66] method from in class reactor.core.publisher.Flux<T> cannot be applied to given types;
[ERROR] required: org.reactivestreams.Publisher<? extends T>
[ERROR] found: org.reactivestreams.Publisher<capture#1 of ?>
[ERROR] reason: cannot infer type-variable(s) T,T
[ERROR] (argument mismatch; org.reactivestreams.Publisher<capture#2 of ?> cannot be converted to org.reactivestreams.Publisher<? extends capture#2 of ?>)
[ERROR] -> [Help 1]
[ERROR] 
```

--- a/spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/ObservableToMessageChannelResultAdapter.java
+++ b/spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/ObservableToMessageChannelResultAdapter.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 the original author or authors.
+ * Copyright 2016-2017 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -16,6 +16,7 @@
 
 package org.springframework.cloud.stream.reactive;
 
+import org.reactivestreams.Publisher;
 import reactor.core.publisher.Flux;
 import rx.Observable;
 import rx.RxReactiveStreams;
@@ -27,6 +28,7 @@ import org.springframework.util.Assert;
 /**
  * A {@link StreamListenerResultAdapter} from an {@link Observable}
  * return type to a bound {@link MessageChannel}.
+ *
  * @author Marius Bogoevici
  */
 public class ObservableToMessageChannelResultAdapter
@@ -47,7 +49,7 @@ public class ObservableToMessageChannelResultAdapter
 	}
 
 	public void adapt(Observable<?> streamListenerResult, MessageChannel bindingTarget) {
-		this.fluxToMessageChannelResultAdapter.adapt(Flux.from(RxReactiveStreams.toPublisher(streamListenerResult)),
-				bindingTarget);
+		Publisher<?> adaptedPublisher = RxReactiveStreams.toPublisher(streamListenerResult);
+		this.fluxToMessageChannelResultAdapter.adapt(Flux.from(adaptedPublisher), bindingTarget);
 	}
 }

